src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -96,8 +96,13 @@ LFORTRAN_API void _lfortran_random_number(int n, double *v)
     }
 }
 
-LFORTRAN_API double _lfortran_random()
+LFORTRAN_API double _lfortran_random(double seed)
 {
+    if (seed == 0) {
+        srand((unsigned int)clock());
+    } else {
+        srand(seed);
+    }
```

Comment:
It seems the definition of `srand` is:
```
void srand( unsigned seed );
```
So `seed` must be of type `unsigned`.

This `srand` should be split into a dedicated function `_lfortran_init_random(unsigned seed)`. This `init_random()` function should only be called once, not for every `random` call. So we should expose it to the user via our runtime library somehow.

Ideally we would emulate what numpy or CPython already does. We would need to investigate.

Reply:
Thanks for the feedback. I refactored the code using a logic similar to [CPython's](https://github.com/python/cpython/blob/80f958529bc5c466add532df4c8dd314995f689f/Lib/random.py#L135). They used classes, but that would make the code way more complex.